src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1660,22 +1660,35 @@ def show_resource(cmd, resource_ids=None, resource_group_name=None,
 # pylint: disable=unused-argument
 def delete_resource(cmd, resource_ids=None, resource_group_name=None,
                     resource_provider_namespace=None, parent_resource_path=None, resource_type=None,
-                    resource_name=None, api_version=None, latest_include_preview=False):
+                    resource_name=None, api_version=None, latest_include_preview=False, tag=None):
     """
     Deletes the given resource(s).
     This function allows deletion of ids with dependencies on one another.
     This is done with multiple passes through the given ids.
     """
-    parsed_ids = _get_parsed_resource_ids(resource_ids) or [_create_parsed_id(cmd.cli_ctx,
-                                                                              resource_group_name,
-                                                                              resource_provider_namespace,
-                                                                              parent_resource_path,
-                                                                              resource_type,
-                                                                              resource_name)]
+    if tag is not None:
+        resources = list_resources(cmd, resource_group_name, resource_provider_namespace, resource_type, resource_name, tag, None)
+        from msrestazure.tools import parse_resource_id
+        # TODO : maybe add a check Resource class in order to access to id (isinstanceof(Resource)) ?
+        resources = [parse_resource_id(resource.id) for resource in resources]
+        parsed_ids = [_create_parsed_id(cmd.cli_ctx,
+                                        resource.get('resource_group', None),
+                                        resource.get('namespace', None),
+                                        resource.get('resource_parent', None),
+                                        resource.get('resource_type', None),
+                                        resource.get('resource_name', None)) for resource in resources]
+    else:
+        parsed_ids = _get_parsed_resource_ids(resource_ids) or [_create_parsed_id(cmd.cli_ctx,
+                                                                                  resource_group_name,
+                                                                                  resource_provider_namespace,
+                                                                                  parent_resource_path,
+                                                                                  resource_type,
+                                                                                  resource_name)]
```

Comment:
When `tag` and `resource_ids` are passed in at the same time, the `resource_ids` will be ignored. Is this in line with the expectation?
If `tag` and `resource_ids` cannot be passed in at the same time, do we need to consider throwing an exception for this situation